### PR TITLE
Remove ability to deploy to dev/qa/sandbox from GH

### PR DESCRIPTION
**Pull request by msnwatson**
## What was the problem?

Associated tickets or Slack threads:
- #3041 

## How to test this PR

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).

### CODE REVIEW

File: .github/workflows/update-deployment.yml [hunk 1]
```diff
@@ -10,8 +10,6 @@ on:
         default: 'dev'
```
Comment (brostk) [inline .github/workflows/update-deployment.yml]: Default should get changed to 'sandbox' I imagine.

**Review by brostk — CHANGES_REQUESTED** [review]

File: .github/workflows/update-deployment.yml (hunk 1 above)
Comment (lisac) [reply .github/workflows/update-deployment.yml]: Wouldn't have spotted that, but I agree.

**Review by lisac — COMMENTED** [review]
question: are rabbitmq and db-init in scope of this work? If so, i think those are managed in these separate files? 
[update-deployment-platform.yml](https://github.com/department-of-veterans-affairs/abd-vro/blob/msnwatson-patch-1/.github/workflows/update-deployment-platform.yml)
[update-deployment-db-init.yml](https://github.com/department-of-veterans-affairs/abd-vro/blob/msnwatson-patch-1/.github/workflows/update-deployment-db-init.yml)

**Comment by msnwatson** [comment]
> question: are rabbitmq and db-init in scope of this work? If so, i think those are managed in these separate files? [update-deployment-platform.yml](https://github.com/department-of-veterans-affairs/abd-vro/blob/msnwatson-patch-1/.github/workflows/update-deployment-platform.yml) [update-deployment-db-init.yml](https://github.com/department-of-veterans-affairs/abd-vro/blob/msnwatson-patch-1/.github/workflows/update-deployment-db-init.yml)

@lisac  I'm actually going to broaden this PR now that we have the new partner team process in place

**Comment by lisac** [comment]
@msnwatson i've just now put up for review the removal of the dev/qa deployment pipelines for partner apps (https://github.com/department-of-veterans-affairs/abd-vro/pull/3355) - if you need to clobber that, just let me know. 


edit: closing 3355 in favor of this PR.

**Review by lisac — APPROVED** [review]

**Pull request merged**